### PR TITLE
FAI-14374 Fail check connection when there are no orgs to fetch

### PR DESCRIPTION
**Pull request by ypc-faros**
Screenshot 2024-12-11 at 2 56 17 PM
Screenshot 2024-12-11 at 2 58 36 PM

### CODE REVIEW

File: sources/github-source/src/github.ts [hunk 1]
```diff
@@ -182,6 +184,17 @@ export abstract class GitHub {
       cfg.authentication.type === 'token'
         ? await GitHubToken.instance(cfg, logger)
         : await GitHubApp.instance(cfg, logger);
+
+    if (cfg.run_mode !== 'EnterpriseCopilotOnly') {
```
Comment (matiaslcoulougian) [inline sources/github-source/src/github.ts:188]: nit: RunMode.EnterpriseCopilotOnly

**Review by matiaslcoulougian — APPROVED** [review]
LGTM

File: sources/github-source/src/org-repo-filter.ts [hunk 2]
```diff
@@ -101,19 +101,16 @@ export class OrgRepoFilter {
       if (!this.filterConfig.organizations) {
         visibleOrgs.forEach((org) => {
           const lowerOrg = toLower(org);
-          if (!this.filterConfig.excludedOrganizations?.has(lowerOrg)) {
-            organizations.add(lowerOrg);
+          if (this.filterConfig.excludedOrganizations?.has(lowerOrg)) {
+            this.logger.info(`Skipping excluded organization ${lowerOrg}`);
```
Comment (dbruno21) [inline sources/github-source/src/org-repo-filter.ts:105]: Is this needed? Thinking that for lot of excluded orgs might be too much

**Review by dbruno21 — APPROVED** [review]
LGTM
Have you done a manual e2e test with a fine-grained token?

File: sources/github-source/src/github.ts [hunk 3]
```diff
@@ -2051,11 +2064,40 @@ export class GitHubToken extends GitHub {
...
+    // so if we get to this point, we're possibly using a fine-grained token.
+    // In order to determine which orgs are visible, check visible repos and track their orgs
+    const seenOrgs = new Set<string>();
+    const reposIter = this.baseOctokit.paginate.iterator(
+      this.baseOctokit.repos.listForAuthenticatedUser,
+      {
+        per_page: this.pageSize,
+        type: 'all',
+      }
+    );
```
Comment (dbruno21) [inline sources/github-source/src/github.ts:2090]: https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#list-repositories-for-the-authenticated-user
Maybe we should filter out here by affiliation = organization_member

Comment (dbruno21) [reply sources/github-source/src/github.ts:2090]: Anyway, using a fine-grained token I don't know what to expect until testing it manually

Comment (ypc-faros) [reply sources/github-source/src/github.ts]: On a second read, `affiliation=organization_member` seems actually more correct

**Comment by ypc-faros** [comment]
> LGTM Have you done a manual e2e test with a fine-grained token?

Done

**Review by dbruno21 — APPROVED** [review]

**Pull request merged**